lib/src/client.dart

```diff
@@ -68,15 +68,15 @@ class Client extends Object with event.Emitter {
     });
   }
 
-  Stream get onOpen => this["open"];
-  Stream get onMessage => this["message"];
-  Stream get onClose => this["close"];
-  Stream get onHeartbeat => this["heartbeat"];
+  Stream<event.Event> get onOpen => getEventStream<event.Event>('open');
+  Stream<MessageEvent> get onMessage => getEventStream<MessageEvent>('message');
+  Stream<CloseEvent> get onClose => getEventStream<CloseEvent>('close');
+  Stream<event.Event> get onHeartbeat => getEventStream<event.Event>('heartbeat');
 
   void close([int code, String reason]) {
     if (_transport != null) {
       if (_transport is WebSocketTransport) {
-        _transport.doClose(code, reason);
+        _transport.doClose(code ?? 1000, reason);
```

Comment:
This change seems like it could use test coverage

Reply:
added